src/moisture_sensor.rs

```diff
@@ -1,49 +1,65 @@
 use std::{
-    cell::RefCell,
     collections::{HashMap},
-    rc::Rc,
     sync::{Arc, Mutex},
 };
 
-use micro_rdk::DoCommand;
-
-use micro_rdk::common::{
-    analog::AnalogReader,
-    config::ConfigType,
-    registry::{self, ComponentRegistry, Dependency, RegistryError},
-    sensor::{
-        GenericReadingsResult, Readings, Sensor, SensorResult, SensorT, SensorType, TypedReadingsResult,
+use micro_rdk::{
+    common::{
+        analog::AnalogReaderType,
+        board::Board,
+        config::ConfigType,
+        registry::{ComponentRegistry, Dependency, RegistryError, get_board_from_dependencies },
+        sensor::{
+            GenericReadingsResult, Readings, Sensor, SensorResult, SensorT, TypedReadingsResult, SensorError, SensorType
+        },
+        status::{Status, StatusError},
     },
-    status::Status,
+    DoCommand,
 };
 
-#[derive(DoCommand)]
-pub struct MoistureSensor(Rc<RefCell<dyn AnalogReader<u16, Error = anyhow::Error>>>);
+//pub struct MoistureSensor{reader : dyn AnalogReader<u16,Error = SensorType>}
 
-pub fn register_model(registry: &mut ComponentRegistry) -> anyhow::Result<(), RegistryError> {
-    registry.register_sensor("moisture", &MoistureSensor::from_config)?;
-    log::info!("moisture sensor registration ok");
+pub fn register_model(registry: &mut ComponentRegistry) -> Result<(), RegistryError> {
```

Comment:
In the other review, I believe you renamed this from `register_model` to `register_models`: https://github.com/viam-labs/micro-rdk-esp32-sensor-examples/pull/6/files#diff-a019dea011c6ec55007918304246cb63a3292f4670e1ef259bf65c6cd6a48672R23 for the per-type registration. Let's do it here too then.